src/pages/index.js

```diff
@@ -62,27 +108,135 @@ const editProfileModal = new PopupWithForm(
 );
 editProfileModal.setEventListeners();
 
+const editAvatarPopup = new PopupWithForm(
+  "#avatar-edit-modal",
+  handleAvatarEditSubmit
+);
+editAvatarPopup.setEventListeners();
+
+const deleteCardPopup = new PopupWithConfirmation("#confirm-delete-modal");
+deleteCardPopup.setEventListeners();
+
 function createCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", openImagePreview);
-  return cardElement.getView();
+  console.log("Creating card with data:", cardData);
+  const cardElement = new Card(
+    cardData,
+    "#card-template",
+    openImagePreview,
+    handleLikeCard,
+    handleDeleteCard
+  );
+  const view = cardElement.getView();
+  console.log("Card element created:", {
+    id: cardElement.getCardId(),
+    element: view,
+  });
+  return view;
+}
+
+function handleDeleteCard(card) {
+  deleteCardPopup.setAction(() => {
+    api
+      .deleteCard(card.getCardId())
+      .then(() => {
+        card.handleDeleteCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => console.error("Error deleting card:", err));
+  });
+  deleteCardPopup.open(card);
+}
+
+function handleLikeCard(card) {
+  if (card.isLiked) {
+    return api
+      .removeLike(card.getCardId())
+      .then(() => {
+        card.setCardLikes(false);
+      })
+      .catch((err) => console.error("Failed to dislike card:", err));
+  } else {
+    return api
+      .addLike(card.getCardId())
+      .then(() => {
+        card.setCardLikes(true);
+      })
+      .catch(console.error);
+  }
 }
 
 function handleProfileEditSubmit(formValues) {
-  userInfo.setUserInfo({
+  const apiData = {
     name: formValues.title,
     about: formValues.description,
-  });
-  editProfileModal.close();
+  };
+
+  editProfileModal.renderLoading(true);
+  api
+    .setUserInfo(apiData)
+    .then((data) => {
+      console.log("API Response:", data);
+      const currentUserInfo = userInfo.getUserInfo();
+
+      userInfo.setUserInfo({
+        name: data.name,
+        about: data.about,
+        avatar: currentUserInfo.avatar,
+      });
+      editProfileModal.close();
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+    })
+    .finally(() => {
+      editProfileModal.renderLoading(false);
+    });
 }
 
+profileImageContainer.addEventListener("click", () => {
+  document.querySelector("#avatar-edit-form").reset();
```

Comment:
You should ** clear ** the inputs only after **submitting** a card rather than when you open or close popups because if you unexpectedly close the popup, you will be upset if the inputs are empty when you open it once more, and you’ll have to type the data again.